src/mococrw/util.h

```diff
@@ -32,6 +33,28 @@ namespace utility
 template <class T>
 using SharedPtrTypeFromUniquePtr = std::shared_ptr<typename T::element_type>;
 
+template <class T>
+std::vector<T> bytesFromFile(const std::string &filename)
+{
+    static_assert(sizeof(T) == sizeof(char), "bytesFromFile only works with 1 byte data types");
+
+    std::ifstream file{filename};
+    if (!file.good()) {
+        std::string errorMsg{"Cannot load key/certificate from file "};
+        errorMsg = errorMsg + filename;
+        throw std::runtime_error(errorMsg);
+    }
+
+    file.seekg(0, std::ios::end);
+    auto size = file.tellg();
+    file.seekg(0, std::ios::beg);
+
+    std::vector<T> buffer;
+    buffer.resize(size);
+    file.read(reinterpret_cast<char *>(buffer.data()), size);
+    return buffer;
+}
```

Comment:
from the error message and content, I'm not sure whether it's a good idea to put this in the public API

it does look like something that ought to be used only internally by mococrw

Reply:
hmm. Jep. Not an easy decision.
In the past all the examples had the keys stored in the source file. For dilithium this was omitted as the keys are really big. Instead the keys are stored in files and read from disk. However this requires some function to read data from disk. The only other example reading data from files is the cert test. However there are functions which reads the data directly from disk.
To keep the interface for dilithium the same as for the other asymmetric keys, it was decided to make this function public.

And if this function is used, the namespace mococrw:utility has to be used. So the programmer should expect a crypto related error message. ;-)